packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.controller.js

```diff
@@ -188,6 +194,46 @@ export default class TelecomPackMigrationOffersCtrl {
     );
   }
 
+  initializeGtrOptions() {
+    this.process.migrationOffers.result.offers = this.process.migrationOffers.result.offers.map(
+      (offer) => {
+        const gtrOptionsMap = Object.values(offer.options)
+          .map((option) => {
+            if (option.name.startsWith('gtr_')) {
+              set(
+                option,
+                'label',
+                this.$translate.instant(
+                  `telecom_pack_migration_${option.name}`,
+                  {
+                    price: option.optionalPrice.text,
+                  },
+                ),
+              );
+              return option;
+            }
+            return undefined;
+          })
+          .filter((el) => el !== undefined);
```

Comment:
**nitpick:**  Since the need is to filter the options which starts with 'gtr' and then add a label. The combination of `filter + map` results in more readability than `map + filter`. Please try out this suggestion and it works.
```suggestion
        const gtrOptionsMap = Object.values(offer.options)
           .filter((el) => el.name.startsWith('gtr_'))
          .map((option) => ({
            ... option,
            label: this.$translate.instant(
                  `telecom_pack_migration_${option.name}`,
                  {
                    price: option.optionalPrice.text,
                  },
                )
          }));
```